mailbox/conndata.go

```diff
@@ -0,0 +1,163 @@
+package mailbox
+
+import (
+	"crypto/sha512"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lightningnetwork/lnd/keychain"
+)
+
+// ConnectionData is the interface that needs to be defined by any object that
+// the Brontide Noise Machine can use to fetch and set connection data.
+type ConnectionData interface {
+	LocalKey() keychain.SingleKeyECDH
+	RemoteKey() *btcec.PublicKey
+	PassphraseEntropy() []byte
+	AuthData() []byte
+	SetRemote(key *btcec.PublicKey) error
+	SetAuthData(data []byte) error
+}
+
+var _ ConnectionData = (*ConnData)(nil)
+
+// ConnData manages all the items needed for establishing a noise connection
+// to a peer. This is an implementation of the ConnectionData interface with
+// some extra methods that are customised for our specific usage of the Noise
+// Machine and the underlying connection.
+type ConnData struct {
+	localKey          keychain.SingleKeyECDH
+	remoteKey         *btcec.PublicKey
+	passphraseEntropy []byte
+	authData          []byte
+
+	onRemoteStatic func(key *btcec.PublicKey) error
+	onAuthData     func(data []byte) error
+
+	mu sync.RWMutex
+}
+
+// NewConnData creates a new ConnData instance.
+func NewConnData(localKey keychain.SingleKeyECDH, remoteKey *btcec.PublicKey,
```

Comment:
Cool, this new struct should also help readability throughout the codebase quite a lot :tada: